### ISSUE TITLE
[explicit-member-accessibility]: add support for tslint member-access config options

### ISSUE THREAD

**Issue by Hotell** [typescript-eslint/typescript-eslint#214]
TSlint [member-access](https://palantir.github.io/tslint/rules/member-access/) supports following config:

- "no-public" forbids public accessibility to be specified, because this is the default.
- "check-accessor" enforces explicit visibility on get/set accessors
- "check-constructor" enforces explicit visibility on constructors
- "check-parameter-property" enforces explicit visibility on parameter properties

Currently this rule follows only all or nothing rule. It would be nice to implement at least `no-public` config to prevent devs to use `public` which is a default in TS.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-member-accessibility": ["error", "no-public"]
  }
}
```

```TS
class Test {
 // Expect lint error
 public render(){
   return `<div>Hello</div>`
 }

 private handleClick = () => { /*...*/ }
}
```

**Comment by bradzacher** [comment]
I don't agree with the `no-public` option in the context of the rule as it is.

If I write out a class and do not annotate any member accessibility, then I will get no lint errors.
It turns it from a rule which forces you to annotate every method, and reminds you if you don't, to a rule which purely forbids the `public` keyword.

In order for us to implement that setting, we would have to rename the rule to `member-accessibility`, as it's no longer about being explicit.

The other settings definitely have merit in the context of the rule right now though.

**Comment by JamesHenry** [comment]
@bradzacher I agree it doesn't fit well with the current name of the rule (vs TSLint's name), but the use-case is valuable when you take the approach of not using `public` and want to enforce it consistently.

This:

```ts
class Foo {
  bar: string;
  baz: string;
  public: qux: string;
}
```

Would be allowable currently in a codebase that didn't want to use `public`, but would be a desirable thing to catch

**Comment by gavinbarron** [comment]
I've been taking a short look at this rule while you two have been chatting here and considering how this might wind up being implemented and the interplay between the options outlined above.

The `no-public` options is pretty easy to implement and I have a spike that handles that vs the current `all` implementation. 

It seems to me that there is a degree of mutual exclusivity in the options. e.g. It don't makes sense to allow a user to specify both no-public and check-constructor. Does that gel with your thinking @JamesHenry?

**Comment by bradzacher** [comment]
That was my thinking - the config is a bit confusing as is (what does it mean to have no-public on and check-accessor on?).

Off the top of my head I'd expect the config to look something like this (we do this for a number of our rules)

```TS
type Override = boolean | { // false = don't check, true = check and use base config, object = override base config 
  noPublic?: boolean;
};
type Options = [
  {
    noPublic?: boolean;
    overrides?: {
      accessor?: Override;
      constructor?: Override;
      method?: Override;
      parameterProperty?: Override;
    }
  }
]
```

**Comment by gavinbarron** [comment]
Yeah, that's making sense to me. 

Are there any guidelines to how to contribute to this project?
I'll admit I was jazzed when the blog post about TypeScript adopting ESLint as the preferred tools came out and I'd like to help make this project as successful as possible for people migrating from TSLint, of which I am a huge fan.

**Comment by gavinbarron** [comment]
@Hotell I have a question for you on the parameter-properties.
Is this required to have a configuration option here given that the `no-parameter-properties` rule allows users to set the allowed accessibility options? If so can you give a use case and description of how the rule is supposed to work and interplay with `no-parameter-properties`?

Thanks

**Comment by kachkaev** [comment]
I guess this can be closed given that #322 is merged
